### ISSUE TITLE
Make groups labelled and visible

### ISSUE THREAD

**Issue by lbovet** [lbovet/archix#2]

**Comment by mrkdev** [comment]
I do not think that the group concept is sufficient.
My requirements are:
- group any elements (instances, hosts) within a given context
- nested contexts

Examples:
- Group on-prem components wihin an on-premise context, off-prem components within an other context.
- Whithin the on-prem context, elements that belong to specific project/business domains must be grouped to.

I propose the creation of an element 'Context'.
- Contexts have a label
- Contexts can contain instances and hosts
- Contexts can contain other contexts
- Contexts are visualized using a grey, dotted line

**Comment by mrkdev** [comment]
@lbovet: are you able to provide such feature soon? Otherwise I will fork and create it myself.

**Comment by lbovet** [comment]
Here is what I propose. We change the "group" feature this way:
1. Instead of having a GroupId class, we use have a Group class.
1. Instead of specifying it in the constructor of Host/Instance, it becomes a method "in" of a common super-class of Node, Instance and Group.
1. Groups can take a label in the constructor. This makes them visible. Otherwise they are invisible and used only to layout things together.
1. They are displayed with a dashed grey line

Example:
```
Group cloud = new Group();
Group domain = new Group("domain").in(cloud);
Host host = new Host().in(domain);
Instance gateway = new Instance("gateway").on(host);
```
Do you agree? I won't be able to work on this before next week. If you do it, just inform me.

**Comment by mrkdev** [comment]
Yes, sounds great!

**Comment by lbovet** [comment]
Started work on nestable groups: https://github.com/lbovet/archix/tree/groups

**Comment by lbovet** [comment]
See the PR https://github.com/lbovet/archix/pull/3
I will merge after testing our work diagrams.

**Comment by lbovet** [comment]
Fixed by PR #3
